p2p/src/channels/rpc/p2p_channels_rpc_reducer.rs

```diff
@@ -31,35 +58,96 @@ impl P2pChannelsRpcState {
                         last_responded: redux::Timestamp::ZERO,
                     },
                 };
+                Ok(())
             }
-            P2pChannelsRpcAction::RequestSend { id, request, .. } => {
-                let Self::Ready { local, .. } = self else {
-                    return;
+            P2pChannelsRpcAction::RequestSend {
+                id,
+                request,
+                on_init,
+                ..
+            } => {
+                let Self::Ready { local, .. } = rpc_state else {
+                    bug_condition!(
+                        "Invalid state for `P2pChannelsRpcAction::RequestSend`, state: {:?}",
+                        rpc_state
+                    );
+                    return Ok(());
                 };
                 *next_local_rpc_id += 1;
                 *local = P2pRpcLocalState::Requested {
                     time: meta.time(),
                     id: *id,
                     request: request.clone(),
                 };
+
+                let dispatcher = state_context.into_dispatcher();
+
+                #[cfg(feature = "p2p-libp2p")]
+                if is_libp2p {
+                    if let Some((query, data)) =
```

Comment:
Under what conditions is this None?

Reply:
Only if request is `P2pRpcRequest::Snark`